packages/components/src/interfaces/complex/CollectionBlock.ts

```diff
@@ -0,0 +1,74 @@
+import type { Static } from "@sinclair/typebox"
+import { Type } from "@sinclair/typebox"
+
+import type { ProcessedCollectionCardProps } from "~/interfaces/internal/CollectionCard"
+import type { IsomerSiteProps, LinkComponentType } from "~/types"
+import { REF_INTERNAL_HREF_PATTERN } from "~/utils/validation"
+
+export const COLLECTION_BLOCK_TYPE = "collectionblock"
+
+export const CollectionBlockSchema = Type.Object(
+  {
+    type: Type.Literal(COLLECTION_BLOCK_TYPE, {
+      default: COLLECTION_BLOCK_TYPE,
+    }),
+    collectionReferenceLink: Type.String({
+      title: "Collection",
+      description: "The collection to display pages from",
+      format: "link",
+      pattern: REF_INTERNAL_HREF_PATTERN,
+    }),
```

Comment:
issue: hmm using the reference link means this is only available for Studio sites and not GitHub ones that we are still supporting. Should we support using permalinks as well?

Reply:
updated to allow legacy too https://github.com/opengovsg/isomer/pull/1250/commits/6fc1dde274642120e563e3d5c17b8d477cc92715